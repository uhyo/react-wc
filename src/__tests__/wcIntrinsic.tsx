import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";
import { wcIntrinsic } from "..";

describe("wcIntrinsic", () => {
  describe("supported", () => {
    it("section: no slot", () => {
      const Hello = wcIntrinsic({
        element: "section",
        shadowHtml: `
            <style>
              div {
                font-size: 100px;
              }
            </style>
            <div>Hello</div>
          `,
      });

      render(<Hello />);

      const el = document.getElementsByTagName("section")[0];

      expect(el.shadowRoot?.innerHTML).toMatchSnapshot();
      expect(el.innerHTML).toMatchSnapshot();
    });
  });
  describe("not supported", () => {
    it("input", () => {
      const Input = wcIntrinsic({
        element: "input",
        shadowHtml: `
            <style>
              :host {
                background-color: yellow;
              }
            </style>
          `,
      });

      expect(() => render(<Input type="password" />)).toThrow();
    });
  });
});
