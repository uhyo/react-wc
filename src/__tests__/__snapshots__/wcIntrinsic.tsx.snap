// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wcIntrinsic section: no slot 1`] = `
"
          <style>
            div {
              font-size: 100px;
            }
          </style>
          <div>Hello</div>
        "
`;

exports[`wcIntrinsic section: no slot 2`] = `""`;

exports[`wcIntrinsic supported section: no slot 1`] = `
"
            <style>
              div {
                font-size: 100px;
              }
            </style>
            <div>Hello</div>
          "
`;

exports[`wcIntrinsic supported section: no slot 2`] = `""`;
